src/main/resources/assets/opencomputers/loot/OpenOS/lib/rc.lua

```diff
@@ -39,14 +53,47 @@ function rc.unload(name)
   loaded[name] = nil
 end
 
-local function rawRunCommand(name, cmd, args, ...)
+local function rawRunCommand(conf, name, cmd, args, ...)
   local result, what = rc.load(name, args)
   if result then
-    if type(result[cmd]) == "function" then
-      result, what = xpcall(result[cmd], debug.traceback, ...)
-      if result then
+    if not cmd then
+      io.output():write("Commands for service " .. name .. "\n")
+      for command, val in pairs(result) do
+        if type(val) == "function" then
+          io.output():write(tostring(command) .. " ")
+        end
+      end
+      return true
+    elseif type(result[cmd]) == "function" then
+      res, what = xpcall(result[cmd], debug.traceback, ...)
+      if res then
         return true
       end
+    elseif cmd == "restart" and type(result["stop"]) == "function" and type(result["start"]) == "function" then
```

Comment:
Maybe also test for the non-existance of a `restart` command here? Maybe a user wants to have control over restarting that is different from simply stopping and starting again.